src/Parsers/ASTSelectWithUnionQuery.h

```diff
@@ -17,6 +16,21 @@ class ASTSelectWithUnionQuery : public ASTQueryWithOutput
     ASTPtr clone() const override;
     void formatQueryImpl(const FormatSettings & settings, FormatState & state, FormatStateStacked frame) const override;
 
+    enum class Mode
+    {
+        Unspecified,
+        ALL,
+        DISTINCT
+    };
+
+    using UnionModes = std::vector<Mode>;
+
+    Mode union_mode;
+
+    UnionModes list_of_modes;
```

Comment:
What do you think about having a single mode for UNION? It is a binary operator, so lumping several of them together into a single structure feels somewhat weird. Later, at the Interpreter stage, we move from binary to N-ary, but still have one mode per union.

Reply:
I am also concerned that "multiple unions per AST" might become problematic when we move to ANTLR grammar and we'll have to redo it anyway -- @abyss7 maybe you have some comments about what is the best representation for UNION ASTs?

Reply:
> What do you think about having a single mode for UNION? It is a binary operator, so lumping several of them together into a single structure feels somewhat weird. Later, at the Interpreter stage, we move from binary to N-ary, but still have one mode per union.

`list_of_modes` was used in parse phase and normalization phase. After normalization, an union list has only one union type, which used in interpretation phase.

Reply:
So this object can represent two logically different structures, depending on the execution phase? Well, never mind, let's merge like this and rewrite when switching to the new grammar. If we only have single union type in interpretation phase, this is good enough.

Reply:
Ok, thanks for carefully review!